Macos compile

gleicon opened the pull request:
Fixed old malloc and memalign issues to compile on mac os high sierra + xcode. Still the binary format looks weird and can't execute the testsuite and won't pass file/lipo as a valid Mach O binary.

<img width="637" alt="screenshot 2018-07-25 21 13 52" src="https://user-images.githubusercontent.com/38321/43234199-16ef0d9e-9051-11e8-9a13-843f6ebab9f7.png">

lpereira commented on src/lib/queue.c:
```diff
@@ -57,11 +56,20 @@
 
     size = lwan_nextpow2(size);
+    #ifdef __APPLE__
+    ret = posix_memalign(q->buffer, sizeof(void *), (1 + size) * sizeof(void *));
```
I think we can get rid of the `#ifdef` and just use `posix_memalign()` everywhere.  Also, the first parameter should be `(void **)&q->buffer`, otherwise it's just not going to work.

lpereira commented:
Also, it not generating a Mach-O binary is indeed weird. I was not able to reproduce this.

gleicon commented:
cleaned up the memalign ifdef/endif section, couldn't figure the Mach-O binary issue, added some info to #194, I'll try some of the things at #170 (as the original error message is Killed 9 and I got this Malformed Mach-o thing from lldb/dtruss).

lpereira commented:
Merged with changes, thanks!

However, I can't reproduce the build issue on High Sierra:

```
Lwans-Mac-mini:b lwan$ file ./src/samples/clock/clock
./src/samples/clock/clock: Mach-O 64-bit executable x86_64
Lwans-Mac-mini:b lwan$ ./src/samples/clock/clock 
140735534912320 lwan-job.c:96 lwan_job_thread_init() Initializing low priority job thread.
140735534912320 lwan-tables.c:40 lwan_tables_init() Uncompressing MIME type table.
140735534912320 lwan.c:383 setup_from_config() Loading configuration file: clock.conf.
140735534912320 lwan-config.c:672 config_open() Could not open configuration file: clock.conf: No such file or directory (error number 2).
140735534912320 lwan-response.c:77 lwan_response_init() Initializing default response.
140735534912320 lwan.c:561 lwan_init_with_config() Initializing lwan web server.
140735534912320 lwan.c:588 lwan_init_with_config() Using 2 threads, maximum 5120 sockets per thread.
140735534912320 lwan-thread.c:548 lwan_thread_init() Initializing threads.
123145524670464 lwan-thread.c:421 thread_io_loop() Starting IO loop on thread #1.
123145525207040 lwan-thread.c:421 thread_io_loop() Starting IO loop on thread #2.
140735534912320 lwan-thread.c:560 lwan_thread_init() IO threads created and ready to serve.
140735534912320 lwan-socket.c:242 lwan_socket_init() Initializing sockets.
140735534912320 lwan-socket.c:167 listen_addrinfo() Listening on http://127.0.0.1:8080.
140735534912320 lwan.c:723 lwan_main_loop() Ready to serve.
```

Another fix was required for Lwan to actually reach `lwan_main_loop()`, but the executable is being built just fine. Also, I've reinstated the macOS build worker, so at least I'll know when the build has been broken there.
